src/main/java/com/insightsystems/symphony/dal/serverlink/SLPSB1008H.java

```diff
@@ -0,0 +1,245 @@
+                    break;
+
+                case "true":
+                case "1":
+                    this.doPost("/ons.cgi?led=" + new String(controlString), "");
+                    if (this.logger.isTraceEnabled())
+                        this.logger.trace("Posting " + this.getHost() + "/ons.cgi?led=" + new String(controlString));
+                    break;
+
+                default:
+                    if (this.logger.isDebugEnabled()){
+                        this.logger.debug("Control value: " + value + " is invalid");
+                    }
+            }
+    }
+
+    @Override
+    public void controlProperties(List<ControllableProperty> list) throws Exception {
+        for( ControllableProperty cp : list){
+            controlProperty(cp);
+        }
+    }
+
+    /**
+     * Performs a regex match on sourceString and returns the first group or an empty string
+     * @param sourceString String to search within
+     * @param regex String regex to search for in sourceString
+     * @return First group match or empty string if not found
+     */
+    private String regexFind(String sourceString,String regex){
+        final Matcher matcher = Pattern.compile(regex).matcher(sourceString);
+        return matcher.find() ? matcher.group(1) : "";
+    }
+
+    /**
+     * Formats outlet name with the outlet number so outlets are ordered correctly on the Symphony portal
+     * @param name Outlet name to be formatted
+     * @param number Number of the outlet
+     * @return Formatted name
+     */
+    private String formatName(String name,int number){
+        String[] split = name.split(" ");
+        for (int i = 0; i < split.length; i++) {
+            split[i] = split[i].substring(0, 1).toUpperCase() + split[i].substring(1).toLowerCase();
+        }
+        return "Outlets#" + "ABCDEFGHIJKLMNOPQRSTUVWXYZ".charAt(number-1) + ": " + String.join(" ",split);
+    }
+
+    /**
+     * Creates a char array used for outlet control and initialises it with a specified value
+     * @param count Number of elements for the array
+     * @param value Initial value for each element of the array.
+     * @return
+     */
+    private char[] createControlArray(int count,char value){
+        char[] returnArray = new char[count];
+        for (int i = 0; i < count; i++){
+            returnArray[i] = value;
+        }
+        return returnArray;
+    }
+
+    private void updateOutletName(int outlet, String name) {
+        if (outletNames.containsKey(outlet)){
+            outletNames.replace(outlet,name);
+        }else{
+            outletNames.put(outlet,name);
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        SLPSB1008H pdu = new SLPSB1008H();
+        pdu.setHost("10.164.69.10");
+        pdu.setPort(80);
+        pdu.setProtocol("http");
+        pdu.setPassword("1234");
+        pdu.setLogin("snmp");
```

Comment:
For the test purposes I'd recommend creating a test class instead of having a main method for it